textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/join/JoinSimilarityTest.java

```diff
@@ -0,0 +1,219 @@
+    
+    public static final KeywordMatchingType conjunction = KeywordMatchingType.CONJUNCTION_INDEXBASED;
+    public static final KeywordMatchingType phrase = KeywordMatchingType.PHRASE_INDEXBASED;
+
+    
+    /*
+     * The annotators @BeforeClass and @AfterClass are used instead of @Before and @After.
+     * 
+     * The difference is that:
+     *   @Before and @After are executed before and after EACH test case.
+     *   @BeforeClass and @AfterClass are executed once before ALL the test begin and ALL the test have finished.
+     * 
+     * We don't want to create and delete the tables on every test case, 
+     *   therefore BeforeClass and AfterClass are better options.
+     *   
+     */
+    @BeforeClass
+    public static void setup() throws TextDBException {
+        // writes the test tables before ALL tests
+        JoinTestHelper.createTestTables();
+    }
+    
+    @AfterClass
+    public static void cleanUp() throws TextDBException {
+        // deletes the test tables after ALL tests
+        JoinTestHelper.deleteTestTables();
+    }
+
+    @After
+    public void clear() throws TextDBException {
+        JoinTestHelper.clearTestTables();
+    }
+
+
+    /*
+     * Tests the Similarity Join Predicate on two similar words:
+     *   Donald J. Trump
+     *   Donald Trump
+     * Under the condition of similarity > 0.8, these two words should match.
+     *
+     */
+    @Test
+    public void test1() throws TextDBException {
+        JoinTestHelper.insertToTable(NEWS_TABLE_OUTER, JoinTestConstants.getNewsTuples().get(0));
+        JoinTestHelper.insertToTable(NEWS_TABLE_INNER, JoinTestConstants.getNewsTuples().get(1));
+
+        String trumpRegex = "[Dd]onald.{1,5}[Tt]rump";
+
+        RegexMatcher regexMatcherInner = JoinTestHelper.getRegexMatcher(JoinTestHelper.NEWS_TABLE_INNER,
+                trumpRegex, JoinTestConstants.NEWS_BODY);
+        RegexMatcher regexMatcherOuter = JoinTestHelper.getRegexMatcher(JoinTestHelper.NEWS_TABLE_OUTER,
+                trumpRegex, JoinTestConstants.NEWS_BODY);
+
+        SimilarityJoinPredicate similarityJoinPredicate = new SimilarityJoinPredicate(JoinTestConstants.NEWS_BODY, 0.8);
+        List<ITuple> results = JoinTestHelper.getJoinDistanceResults(
+                regexMatcherOuter, regexMatcherInner, similarityJoinPredicate, Integer.MAX_VALUE, 0);
+
+        Schema joinInputSchema = Utils.addAttributeToSchema(JoinTestConstants.NEWS_SCHEMA, SchemaConstants.SPAN_LIST_ATTRIBUTE);
+        Schema resultSchema = similarityJoinPredicate.generateOutputSchema(joinInputSchema, joinInputSchema);
+
+        List<Span> resultSpanList = Arrays.asList(
+                new Span("inner_"+JoinTestConstants.NEWS_BODY, 5, 20, trumpRegex, "Donald J. Trump", -1),
+                new Span("outer_"+JoinTestConstants.NEWS_BODY, 18, 30, trumpRegex, "Donald Trump", -1)
+        );
+
+        ITuple resultTuple = new DataTuple(resultSchema,
```

Comment:
"resultTuple" -> "expectedResultTuple".  Same comment for other test cases in this PR.